src/Integration/UserSettingsConfiguration/UserSettingsProvider.cs

```diff
@@ -29,16 +29,22 @@
 namespace SonarLint.VisualStudio.Integration.UserSettingsConfiguration;
 
 [Export(typeof(IUserSettingsProvider))]
+[Export(typeof(IGlobalRawSettingsService))]
+[Export(typeof(ISolutionRawSettingsService))]
 [PartCreationPolicy(CreationPolicy.Shared)]
-internal sealed class UserSettingsProvider : IUserSettingsProvider, IDisposable
+internal sealed class UserSettingsProvider : IUserSettingsProvider, IGlobalRawSettingsService, ISolutionRawSettingsService, IDisposable
 {
     private static readonly object Lock = new();
     private readonly IActiveSolutionTracker activeSolutionTracker;
     private readonly ILogger logger;
     private readonly IGlobalSettingsStorage globalSettingsStorage;
     private readonly ISolutionSettingsStorage solutionSettingsStorage;
     private UserSettings userSettings;
```

Comment:
We could use the word `Cached` in the name of this variable and the 2 below for clarity, for example `globalAnalysisSettingsCached`

Alternatively, group them together in a tuple to indicate that they're updated together, like we did with the file paths in Storages